### PR TITLE
[ISSUE #6108] Optimize command line tools QueryMsgByKey& QueryMsgTraceById

### LINKED ISSUE #6108: Command line tool QueryMsgByKey& QueryMsgTraceById：time range and number of data bars can be specified
Command line function: Query messages by Key. The time range and number of data bars cannot be specified at this time. Fixed returning 64 data and query full messages.In some special scenarios, the requirements cannot be met.

Therefore, you are advised to optimize the logic and specify the time range and number of data items. QueryMsgTraceByIdSubCommand is in the same situation

- After optimization, QueryMsgByKeySubCommand help: 

usage: mqadmin queryMsgByKey [-b <arg>] [-e <arg>] [-h] -k <arg> [-n <arg>] [-num <arg>] -t <arg>
 -b,--beginTimestamp <arg>   Begin timestamp(ms). default:0, eg:1676730526212
 -e,--endTimestamp <arg>     End timestamp(ms). default:Long.MAX_VALUE, eg:1676730526212
 -h,--help                   Print help
 -k,--msgKey <arg>           Message Key
 -n,--namesrvAddr <arg>      Name server address list, eg: '192.168.0.1:9876;192.168.0.2:9876'
 -num,--maxNum <arg>         The maximum number of messages returned by the query, default:64
 -t,--topic <arg>            topic name

- After optimization,QueryMsgTraceByIdSubCommand help: 

usage: mqadmin queryMsgTraceById [-b <arg>] [-e <arg>] [-h] -i <arg> [-n <arg>] [-num <arg>] [-t <arg>]
 -b,--beginTimestamp <arg>   Begin timestamp(ms). default:0, eg:1676730526212
 -e,--endTimestamp <arg>     End timestamp(ms). default:Long.MAX_VALUE, eg:1676730526212
 -h,--help                   Print help
 -i,--msgId <arg>            Message Id
 -n,--namesrvAddr <arg>      Name server address list, eg: '192.168.0.1:9876;192.168.0.2:9876'
 -num,--maxNum <arg>         The maximum number of messages returned by the query, default:64
 -t,--traceTopic <arg>       The name value of message trace topic

### PR DESCRIPTION
**Pull request by socutes**
close #6108

### CODE REVIEW

**Review by mxsm — CHANGES_REQUESTED** [review]
Hi @socutes  CI fails and you can import codestyle into your idea. reference: https://rocketmq.apache.org/docs/contributionGuide/02code-guidelines

**Comment by socutes** [comment]
> https://rocketmq.apache.org/docs/contributionGuide/02code-guidelines

ok. Thanks~

**Comment by socutes** [comment]
@zhouxinyu PTAL

**Review by drpmma — DISMISSED** [review]
LGTM

**Review by echooymxq — COMMENTED** [review]
Hi, thanks your contributions, you should also update the `operation.md` docs.

**Comment by echooymxq** [comment]
https://github.com/apache/rocketmq/pull/5772, Maybe this Pr will close.

**Comment by socutes** [comment]
> #5772, Maybe this Pr will close.

hi, I didn't notice this closed issue. This problem has not been fundamentally solved. The main branch does not have this function.
I think this problem is worth solving.

File: tools/src/main/java/org/apache/rocketmq/tools/command/message/QueryMsgTraceByIdSubCommand.java [hunk 1]
```diff
@@ -47,6 +47,19 @@ public Options buildCommandlineOptions(Options options) {
...
+
+        opt = new Option("b", "beginTimestamp", true, "Begin timestamp(ms). default:0, eg:1676730526212");
+        opt.setRequired(false);
+        options.addOption(opt);
+
+        opt = new Option("e", "endTimestamp", true, "End timestamp(ms). default:Long.MAX_VALUE, eg:1676730526212");
+        opt.setRequired(false);
+        options.addOption(opt);
+
+        opt = new Option("num", "maxNum", true, "The maximum number of messages returned by the query, default:64");
```
[the remark this answers is not in the document]
Comment (socutes) [reply tools/src/main/java/org/apache/rocketmq/tools/command/message/QueryMsgTraceByIdSubCommand.java:59]: I was thinking about this before, and I didn't think of the right shorthand, so I used num.
Your suggestion is very good. I'll revise it.

**Review by zhouxinyu — COMMENTED** [review]
LGTM~

**Review by zhouxinyu — APPROVED** [review]

**Pull request merged**